compliance/collection/main.go

```diff
@@ -181,6 +175,42 @@ func initializeStream(ctx context.Context, cli sensor.ComplianceServiceClient) (
 	return client, config, nil
 }
 
+func manageNodeScanLoop(ctx context.Context, client sensor.ComplianceService_CommunicateClient, scanner nodescanv2.NodeScanner) {
+	t := time.NewTicker(env.NodeScanInterval.DurationSetting())
```

Comment:
I didn't check that before, but I see that `DurationSetting()` can return 0 or negative durations. The Ticker must get positive >0 duration or it will panic. I think, we need a check that would prevent us from running into problems when users provide trash input through the env variable.

Reply:
Confirmed that a negative value causes a crash: 

```
│ panic: non-positive interval for NewTicker                                                                                                                                                          │
│                                                                                                                                                                                                     │
│ goroutine 26 [running]:                                                                                                                                                                             │
│ time.NewTicker(0xc0007129d8?)                                                                                                                                                                       │
│     time/tick.go:24 +0x10f                                                                                                                                                                          │
│ main.manageNodeScanLoop({0x233f9a0, 0xc000592090}, {0x2345340, 0xc0005234f0}, {0x2319460, 0x369f8e8})                                                                                               │
│     github.com/stackrox/rox/compliance/collection/main.go:179 +0x6a                                                                                                                                 │
│ created by main.main                                                                                                                                                                                │
│     github.com/stackrox/rox/compliance/collection/main.go:244 +0x525                                                                                                                                │
│ Stream closed EOF for stackrox/collector-9qcdt (compliance)
```

Reply:
Thank you for checking this! 
I have implemented a safety check to validate what is put into this variable.